components/org.wso2.carbon.identity.oauth.endpoint/src/main/java/org/wso2/carbon/identity/oauth/endpoint/util/EndpointUtil.java

```diff
@@ -706,47 +737,255 @@ public static String getUserConsentURL(OAuth2Parameters params, String loggedInU
+     * @return  True if user has given consent to all the requested OAuth scopes.
+     * @throws IdentityOAuth2ScopeConsentException
+     * @throws IdentityOAuthAdminException
+     */
+    public static boolean isUserAlreadyConsentedForOAuthScopes(AuthenticatedUser user,
+                                                               OAuth2Parameters oAuth2Parameters)
+            throws IdentityOAuth2ScopeException, IdentityOAuthAdminException, OAuthSystemException {
+
+        List<String> scopesToBeConsented = new ArrayList<>(oAuth2Parameters.getScopes());
+        if (log.isDebugEnabled()) {
+            log.debug("Checking if user has already provided the consent for requested scopes : " +
+                    scopesToBeConsented.stream().collect(Collectors.joining(" ")) + " for client : " +
+                    oAuth2Parameters.getClientId());
+        }
+        // Remove OIDC scopes.
+        scopesToBeConsented.removeAll(getOIDCScopeNames());
+        String userId = getUserIdOfAuthenticatedUser(user);
+        String appId = getAppIdFromClientId(oAuth2Parameters.getClientId());
+        return oAuth2ScopeService.hasUserProvidedConsentForAllRequestedScopes(userId, appId,
+                IdentityTenantUtil.getTenantId(user.getTenantDomain()), scopesToBeConsented);
+    }
+
+    /**
+     * Store consent given for OAuth scopes by the user for the application.
+     *
+     * @param user                      Authenticated user.
+     * @param params                    OAuth2 parameters.
+     * @param overrideExistingConsent   True to override existing consent, otherwise merge the new consent with
+     *                                  existing consent.
+     * @throws OAuthSystemException
+     */
+    public static void storeOAuthScopeConsent(AuthenticatedUser user, OAuth2Parameters params,
+                                              boolean overrideExistingConsent) throws OAuthSystemException {
 
         try {
-            Set<String> allowedScopes = params.getScopes();
-            startTenantFlow(params.getTenantDomain());
+            Set<String> userApprovedScopesSet = params.getConsentRequiredScopes();
+            if (CollectionUtils.isNotEmpty(userApprovedScopesSet)) {
+                if (log.isDebugEnabled()) {
+                    log.debug("Storing user consent for approved scopes : " + userApprovedScopesSet.stream()
+                            .collect(Collectors.joining(" ")) + " of client : " + params.getClientId());
+                }
+                List<String> userApprovedScopes = new ArrayList<>(userApprovedScopesSet);
+                // Remove OIDC scopes.
+                userApprovedScopes.removeAll(getOIDCScopeNames());
+                String userId = getUserIdOfAuthenticatedUser(user);
+                String appId = getAppIdFromClientId(params.getClientId());
+                if (overrideExistingConsent) {
+                    oAuth2ScopeService.addUserConsentForApplication(userId, appId,
```

Comment:
lets add a debug log here

Reply:
Fixed in https://github.com/wso2-extensions/identity-inbound-auth-oauth/pull/1548/commits/6dbf4cf4e588670e742bf004582eb75fae29566e